### DISCUSSION TITLE
Отдельно КПП, Кнопки и ручник.

### DISCUSSION THREAD

**Discussion by lazenyuk-dmitry** [vsulako/AFFBWheel#31] [Ideas]
@vsulako, привет. Появилась такая мысль, может быть можно сделать еще пару проектов которые будут реализовывать функционал КПП, Кнопок и ручника. Например есть проблема шума при отключении осей, а так можно сделать например отдельно КПП на потенциометрах на каком ни будь ProMicro и просто подключать не к базе а отдельно в USB. Что было бы иногда удобно например не нужна тебе КПП сейчас ну просто не подключай и не нужно идти отключать оси в GUI. 

Но тут есть свои плюсы и минусы. Например можно немного разгрузить память например в основной прошивке КПП только на кнопках и какие то базовые возможности. А отдельно уже полный фарш и своя GUI для настройки всего этого дела. 

Но опять же это немного сделает сборку дороже, но может быть можно это реализовать на чем то дешевом? Ну а так будет версия для богатых можно сказать) Я сижу в теме обсуждения самодельных рулей так некоторые покупают себе STM для того чтобы сделать только кнопочки) 

https://github.com/YukMingLaw/ArduinoJoystickWithFFBLibrary вот эта библиотека думаю может помочь в реализации. 

Я бы тоже может попробовал что то сделать такое, выйти дальше своего фронтенда) Но пока времени нету на это. Ну и может думаю тебе будет интересно это да и какие то варианты предложишь ну и вообще обсудить эту идею.

**Comment by vsulako** [comment]
Так ведь давно существуют [MMJoy2](https://sites.google.com/site/mmjoyproject/) для ProMicro/Leonardo и [Freejoy ](https://github.com/FreeJoy-Team/FreeJoy)для BluePill. С их помощью делается любой нужный контроллер, до максимума в 8 осей и до 128 кнопок в любой конфигурации, все настраивается через GUI. Хоть КПП, хоть педали, хоть целая панель из кнопок.
Если бы мне понадобился отдельный контроллер, я бы делал именно на них. Все уже есть готовое, повторять смысла нет.
Вот чего там нет так это обратной связи, а то бы свой проект делать не пришлось.
Касательно цены - ну раз STM32 сейчас достаточно дешевы ([ProMicro](https://aliexpress.ru/item/1005003683743044.html) - под 400р, а [BluePill](https://aliexpress.ru/item/1005003169244854.html) - около 150р, странный перекос, но что поделать), то и я бы для кнопочек брал именно его :) 
Из дешевых вот еще есть [CH552G](https://aliexpress.ru/item/1005005265280621.html) (~120р) с поддержкой USB. Можно было бы сделать, скажем, какой-то аналог MMJoy2/Freejoy для них. С другой стороны, разница в цене с BluePill не такая большая, чтобы этим заморачиваться. 
А вот еще лучше было бы перетащить весь проект на BluePill, там по сравнению с Atmega32U4 и памяти, и быстродействия в разы больше. Но это для меня незнакомая область, надо там все долго изучать, а пока что у меня даже среду разработки чего-то завести не удалось.

**Reply by lazenyuk-dmitry** [reply]
Странно. Я думал stm дороже будет) Да на STM перенести наверно было бы хорошо) И памяти побольше. А может быть даже лучше поискать покруче что то чем Blue Pill? Чтобы памяти побольше и тд? Ты наверно насчет этого уже смотрел инфу я так думаю?) Или 64КБ на все что хочешь хватит для такого проекта и смысла переплачивать нету? Я бы тоже может заказал бы STM да попробовал плюсы поучить) Но это как время будет.

@vsulako ты можешь написать мне в телеграм например? У меня контакты в профиле на гитхаб все есть. А то общаемся с тобой только через гитхаб. Контакты твои не спалю третим лицам, если у тебя такая конфиденциальность) Просто было бы удобнее иногда да и может поможешь с плюсами как я надумаю их учить)
